admin_pages/general_settings/OrganizationSettings.php

```diff
@@ -494,85 +495,16 @@ public function process($form_data = array())
         if ($country instanceof EE_Country) {
             $country->set('CNT_active', 1);
             $country->save();
-            $this->getCountrySubRegions($country);
+            /** @var \EventEspresso\core\services\address\CountrySubRegionDao $countryDao */
+            $countryDao = LoaderFactory::getLoader()->getShared(
+                'EventEspresso\core\services\address\CountrySubRegionDao'
+            );
+            $countryDao->saveCountrySubRegions($country);
```

Comment:
are we not injecting CountrySubRegionDao for back compat reasons (which makes sense if so)?  I wonder if we should still inject `CountrySubRegionDao = null`  and only load if its not present? Would that be a good transition preserving back-compat but keeping dependencies in order?

Reply:
it was more a case of not loading that unless it is needed which would only occur if a country is selected that does not have any states/provinces in the db. So it really only needs to load once in a blue moon. In most cases this may literally only ever need to load once, and that's only if the customer's set country does not have any states. Loading it by default for every other request to the page is totally unnecessary.

Reply:
hmm sure, but all it is doing is constructing it which isn't much different from what we do elsewhere.  I'm actually kinda surprised you didn't inject this.